internal/context.go

```diff
@@ -0,0 +1,274 @@
+package internal
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"strings"
+
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v2"
+)
+
+// ContextHandler maps strings to contexts and keeps track of the current context.
+// It has various methods for manipulating and accessing Contexts.
+type ContextHandler struct {
+	Current  string `yaml:"current-context"`
+	Contexts map[string]*Context
+}
+
+// Context represents a context. As of now it only contains information regarding connections.
+// Meaning: engine url, certificates path and any headers.
+// It also keeps a user's comments regarding the context.
+type Context struct {
+	Engine       string
+	Headers      map[string]string
+	Certificates string
+	Comment      string
+}
+
+var contextFilePath = path.Join(userHomeDir(), ".corectl", "contexts.yml")
+
+func CreateContext(contextName, comment string) string {
+	if contextName == "" {
+		FatalError("\"\" is not a valid context name")
```

Comment:
Maybe empty instead of ""?

Reply:
Changed the phrasing, should be clearer now.